rfq/order.go

```diff
@@ -95,9 +103,17 @@ type AssetSalePolicy struct {
 	// the policy.
 	MaxOutboundAssetAmount uint64
 
+	// CurrentAssetAmountMsat is the total amount that is held currently in
+	// accepted htlcs.
+	CurrentAmountMsat lnwire.MilliSatoshi
+
 	// AskAssetRate is the quote's asking asset unit to BTC conversion rate.
 	AskAssetRate rfqmath.BigIntFixedPoint
 
+	// htlcToAmt maps the unique htlc identifiers to the effective amount
+	// that they carry.
+	htlcToAmt lnutils.SyncMap[string, lnwire.MilliSatoshi]
```

Comment:
So both the `htlcToAmt` and `CurrentAmountMsat` need to be in sync. If we use an atomic value for one and a sync map for the other, then it's still possible for a _read_ from another goroutine to get a different value while we're updating things (e.g. the map was updated but the total not yet).

So I think because we have two separate values, we need a `sync.RWMutex` instead. But then we can use a normal map here at least.

Reply:
> So both the htlcToAmt and CurrentAmountMsat need to be in sync

What do you mean by "in sync"? The `htlcToAmt` is only used to cache the htlc's effective amount when the time to "cancel" it comes (i.e call `UntrackHtlc`)

On the other hand, `CurrentAmountMsat` is incremented/decremented by multiple goroutines, so def agree that we have to guard that.

So could just use a mutex that protects `CurrentAmountMsat` alone